contributors/design-proposals/pod-resource-management.md

```diff
@@ -247,170 +477,12 @@ spec:
-- `BE` pods are treated as lowest priority. So for the `BE` QoS cgroup we set cpu shares to the lowest possible value ie.2. This ensures that the `BE` containers get a relatively small share of cpu time.
-- Also we don't set any quota on the cpu resources as the containers on the `BE` pods can use any amount of free resources on the node.
-- Having memory limit of `BE` cgroup as (Allocatable - summation of memory requests of `G` and `Bu` pods) would result in `BE` pods becoming more susceptible to being OOM killed. As more `G` and `Bu` pods are scheduled kubelet will more likely kill `BE` pods, even if the `G` and `Bu` pods are using less than their request since we will be dynamically reducing the size of `BE` m.limit_in_bytes. But this allows for better memory guarantees to the `G` and `Bu` pods.
-
-## Implementation Plan
-
-The implementation plan is outlined in the next sections.
-We will have a 'experimental-cgroups-per-qos' flag to specify if the user wants to use the QoS based cgroup hierarchy. The flag would be set to false by default at least in v1.5.
-
-#### Top level Cgroups for QoS tiers
-
-Two top level cgroups for `Bu` and `BE` QoS classes are created when Kubelet starts to run on a node. All `G` pods cgroups are by default nested under the `Root`. So we dont create a top level cgroup for the `G` class. For raw cgroup systems we would use libcontainers cgroups manager for general cgroup management(cgroup creation/destruction). But for systemd we don't have equivalent support for slice management in libcontainer yet. So we will be adding support for the same in the Kubelet. These cgroups are only created once on Kubelet initialization as a part of node setup. Also on systemd these cgroups are transient units and will not survive reboot.
-
-#### Pod level Cgroup creation and deletion (Docker runtime)
-
-- When a new pod is brought up, its QoS class is firstly determined.
-- We add an interface to Kubelet's ContainerManager to create and delete pod level cgroups under the cgroup that matches the pod's QoS class.
-- This interface will be pluggable. Kubelet will support both systemd and raw cgroups based __cgroup__ drivers. We will be using the --cgroup-driver flag proposed in the [Systemd Node Spec](kubelet-systemd.md) to specify the cgroup driver.
-- We inject creation and deletion of pod level cgroups into the pod workers.
-- As new pods are added QoS class cgroup parameters are updated to match the resource requests by the Pod.
-
-#### Container level cgroups
-
-Have docker manager create container cgroups under pod level cgroups. With the docker runtime, we will pass --cgroup-parent using the syntax expected for the corresponding cgroup-driver the runtime was configured to use.
-
-#### Rkt runtime
-
-We want to have rkt create pods under a root QoS class that kubelet specifies, and set pod level cgroup parameters mentioned in this proposal by itself.
```

Comment:
I think `rkt` needs to support it.  I think individual container runtime support should be separate from this proposal generally.  Looking in the current code in head, I do not see it being used.